### PR TITLE
add interface for deepmind lab enviroments

### PR DESCRIPTION
**Pull request by witwolf**
many algorithm about exploration using dmlab as test envs, it would be helpful if we want to try these algorithms . and i am try to add merlin examples using dmlab envs mentioned in the paper . 
we do not have to check this pr in if it's not very useful

### CODE REVIEW

File: alf/environments/suite_dmlab.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+def action_discretize(action_spec,
+                      look_left_right_pixels_per_frame=(-20, 20),
+                      look_down_up_pixels_per_frame=(-10, 10),
+                      strafe_left_right=(-1, 1),
+                      move_back_forward=(-1, 1),
+                      fire=(),
+                      jump=(1, ),
+                      crouch=(1, ),
+                      **kwargs):
+    """Discretize action from action_spec
```
Comment (hnyu) [inline alf/environments/suite_dmlab.py]: More details: describe what the discretization does by giving examples, especially what we should expect about the output representation.

File: alf/environments/suite_dmlab.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+                 action_repeat=4,
+                 observation='RGB_INTERLEAVED',
+                 config={},
+                 renderer='software'):
+        """Create an deepmind_lab env
+
+        Args:
+            scene (str): script for the deepmind_lab env. See available script:
+                `https://github.com/deepmind/lab/tree/master/game_scripts/levels`
+            action_repeat (int): the frequency at which the agent experiences the game
```
Comment (hnyu) [inline alf/environments/suite_dmlab.py]: 'frequency' -> 'interval'

File: alf/environments/suite_dmlab.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+        val_max = spec['max']
+        values = config.get(spec['name'], None)
+
+        if values is None:
+            values = list(range(val_min, val_max + 1))
+        elif isinstance(values, str):
+            values = eval(values)
+
+        for value in values:
+            if value < val_min or value > val_max or value == 0:
```
Comment (hnyu) [inline alf/environments/suite_dmlab.py]: why ignoring value == 0?

File: alf/environments/suite_dmlab.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+    def seed(self, seed=None):
+        self._lab.reset(seed=seed)
+
+    def close(self):
+        self._lab.close()
+
+    def render(self, mode='rgb_array', close=False):
+        if mode == 'rgb_array':
+            return self._last_obs
```
Comment (hnyu) [inline alf/environments/suite_dmlab.py]: raise errors in other cases

File: alf/environments/wrappers.py [hunk 5]
```diff
@@ -110,6 +110,47 @@ def __getattr__(self, name):
...
+        super().__init__(env)
+        obs_shape = env.observation_space.shape
+        self.observation_space = gym.spaces.Box(
+            low=0, high=255, shape=list(obs_shape[:-1]) + [1], dtype=np.uint8)
+
+    def observation(self, obs):
+        obs = cv2.cvtColor(obs, cv2.COLOR_RGB2GRAY)
+        return np.expand_dims(obs, -1)
+
+    def __getattr__(self, name):
```
Comment (hnyu) [inline alf/environments/wrappers.py]: You don't need this function

File: alf/environments/wrappers.py [hunk 6]
```diff
@@ -110,6 +110,47 @@ def __getattr__(self, name):
...
+        return obs
+
+
+@gin.configurable
+class FrameGrayScale(gym.ObservationWrapper):
+    """Gray scale image observation"""
+
+    def __init__(self, env):
+        super().__init__(env)
+        obs_shape = env.observation_space.shape
```
Comment (hnyu) [inline alf/environments/wrappers.py]: should check the current obs_shape. Make sure it has no more than three dims.

File: alf/environments/wrappers.py [hunk 7]
```diff
@@ -110,6 +110,47 @@ def __getattr__(self, name):
         return getattr(self.env, name)
 
 
+@gin.configurable
+class FrameResize(gym.ObservationWrapper):
+    def __init__(self, env, width=84, height=84):
+        super().__init__(env)
+        self._width = width
+        self._height = height
+        obs_shape = env.observation_space.shape
```
Comment (hnyu) [inline alf/environments/wrappers.py]: should check the current obs_shape. Make sure it has no more than three dims.

File: alf/environments/suite_dmlab.py (hunk 3 above)
Comment (witwolf) [reply alf/environments/suite_dmlab.py]: valid value for action is   `[min, max]  - {0}`, 0 means idle in this direction

File: alf/environments/suite_dmlab.py [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+class DeepmindLabEnv(gym.Env):
+    metadata = {'render.modes': ['rgb_array']}
+
+    def __init__(self,
+                 scene,
+                 action_repeat=4,
+                 observation='RGB_INTERLEAVED',
+                 config={},
+                 renderer='hardware'):
+        """Create an deepmind_lab env
```
Comment (emailweixu) [inline alf/environments/suite_dmlab.py]: an => a

**Pull request merged**